docs/deployment-guide/migrating.md

```diff
@@ -0,0 +1,63 @@
+# Migration Guide
+
+This guide provides instructions for migrating from the ngrok Ingress Controller chart to the ngrok Operator chart.
+
+**Note: It is recommended you test this upgrade procedure in a non-production environment before upgrading your production environment.**
+
+## Migrating from the ngrok Ingress Controller chart to the ngrok Operator chart
+
+1. Create a new namespace for the ngrok Operator:
+
+   ```shell
+   kubectl create namespace ngrok-operator
+   ```
+
+2. Copy the `ngrok-credentials` secret from the old namespace, usually `ngrok-ingress-controller`, to the new namespace and verify it is correct. Your secret name may be different if you have customized the installation:
+
+   ```shell
+   kubectl get secret ngrok-credentials -n ngrok-ingress-controller -o yaml | sed 's/namespace: ngrok-ingress-controller/namespace: ngrok-operator/' | kubectl apply -n ngrok-operator -f -
+   ```
+
+3. Change the helm ownership of the ngrok CRDS and existing ingress class. Your release name and ingressClass name may be different if you have customized the installation:
+
+   ```shell
+   kubectl get crds | grep "ngrok" | awk '{print $1}' | xargs -I '{}' kubectl annotate crd '{}' 'meta.helm.sh/release-name=ngrok-operator' --overwrite
+   ```
+   ```shell
+   kubectl get crds | grep "ngrok" | awk '{print $1}' | xargs -I '{}' kubectl annotate crd '{}' 'meta.helm.sh/release-namespace=ngrok-operator' --overwrite
+   ```
+   ```shell
+   kubectl get ingressclass | grep "ngrok" | awk '{print $1}' | xargs -I '{}' kubectl annotate ingressclass '{}' 'meta.helm.sh/release-name=ngrok-operator' --overwrite
+   ```
+   ```shell
+   kubectl get ingressclass | grep "ngrok" | awk '{print $1}' | xargs -I '{}' kubectl annotate ingressclass '{}' 'meta.helm.sh/release-namespace=ngrok-operator' --overwrite
```

Comment:
I'm not sure how many people use gateway, but we might need similar instructions for the gateway class cluster level resource

Reply:
Good call, I'll update the instructions to match